feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/checkbox-quantity/configurator-attribute-checkbox-quantity.component.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { Configurator } from '../../../../../core/model/configurator.model';
+import { FormControl } from '@angular/forms';
+import { QuantityUpdateEvent } from '../../../../form/configurator-form.event';
+import { BehaviorSubject, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'cx-configurator-attribute-checkbox-quantity',
+  templateUrl: './configurator-attribute-checkbox-quantity.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorAttributeCheckBoxQuantityComponent
+  implements OnDestroy, OnInit {
+  loading$ = new BehaviorSubject<boolean>(false);
+  quantity = new FormControl(1);
+  private sub: Subscription;
+
+  @Input() value: Configurator.Value;
+  @Output() handleQuantity = new EventEmitter<QuantityUpdateEvent>();
+
+  ngOnInit() {
+    this.quantity.setValue(this.value.quantity);
+
+    this.sub = this.quantity.valueChanges.subscribe(() =>
+      this.onHandleQuantity()
+    );
+  }
+
+  ngOnDestroy() {
+    this.sub.unsubscribe();
+  }
+
+  onHandleQuantity(): void {
+    this.loading$.next(true);
```

Comment:
This means the control is disabled only when an update for the same qty control is pending.
What about updates from other qty controls, or updates from other attribute values? Is the CPQ API capable of handling multiple changes? (Just asking, please check with Sascha or Uli)

Reply:
CPQ can handle only update of one attribute (either value or quantity) or update of one value quantity during one API call

Reply:
And what about multiple, parallel API calls?

Reply:
I believe this should be possible. but i double check with Ana.

Reply:
she confirmed that CPQ will queue the requests for us. so no actions from our side